Sources/BSON/Document.swift

```diff
@@ -9,55 +36,115 @@ public struct Document {
 
     private var _buffer: ByteBuffer
 
+    internal var size: Int {
+        guard let size = self.buffer.getInteger(at: 0, endianness: .little, as: Int32.self) else {
```

Comment:
would you ever expect reading the int32 here to fail?

Reply:
Not currently we should always have the int32 bytes in the buffer. I do have to unwrap and cast the value nonetheless, is there a cleaner syntax for that here?

Reply:
id say we should just fatalError as it would be invalid BSON if this fails